### DISCUSSION TITLE
SingleSelect: Changing width doesn't change popper width

### DISCUSSION THREAD

**Discussion by martinkrulltott** [dhis2/notes#221] [Q&A] [answered]
When a `SingleSelect` changes size, because e.g. more elements were added to the same container, the size of the popper that displays when clicking the `SingleSelect` doesn't change size. Meaning the popper doesn't match the width of the initiator.

As seen in this screenshot:
![image](https://user-images.githubusercontent.com/12590483/107488718-09425b00-6b88-11eb-9bcf-2e7549504be7.png)

Simple example: https://codesandbox.io/s/ui-select-width-xzstj?file=/src/App.js

In Data Visualizer, this issues is appearing in the data dimension selector, like this: 
![image](https://user-images.githubusercontent.com/12590483/107489101-85d53980-6b88-11eb-98c7-9d873cf8cfd5.png)

**Comment by ghost** [comment]
Ok, that's a regression. Would you mind opening a bug report in Jira for that?

**Reply by HendrikThePendric** [reply]
I've looked into this some more and I think this is a new bug, and not technically a regression. We have some code in place to control the menuWidth:
1) The `Select` determines the `menuWidth` in `componentDidMount`
2) The `Popper` has been given the `observeReferenceResize` which controls the menu's width once the Popper is open

In the current case, the width changes *after* the component has mounted and *before* the Popper is showing. So we hadn't catered for that yet. I hope/expect simply calling `setMenuWidth` when the Popper opens is enough.

**Reply by martinkrulltott** [reply]
JIRA ticket: https://jira.dhis2.org/browse/LIBS-119

**Comment by HendrikThePendric** [comment, marked as answer]
Fix is here: https://github.com/dhis2/ui/pull/501

**Reply by HendrikThePendric** [reply]
Resleased as [v6.2.2](https://www.npmjs.com/package/@dhis2/ui/v/6.2.2)
